src/gui/editorwidgets/core/qgsdefaultsearchwidgetwrapper.h

```diff
@@ -0,0 +1,50 @@
+/***************************************************************************
+    qgsdefaultsearchwidgetwrapper.h
+     --------------------------------------
+    Date                 : 21.5.2015
+    Copyright            : (C) 2015 Karolina Alexiou
+    Email                : carolinegr at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSDEFAULTSEARCHWIDGETWRAPPER_H
+#define QGSDEFAULTSEARCHWIDGETWRAPPER_H 
+
+#include "qgseditorwidgetwrapper.h"
+#include <qgsfilterlineedit.h>
+
+
+/**
+ * Wraps a search widget. Default form is just a QgsLineFilterEdit 
+ *
+ */
+
+class GUI_EXPORT QgsDefaultSearchWidgetWrapper : public QgsEditorWidgetWrapper
```

Comment:
I think this does not have to inherit from QgsEditorWidgetWrapper (the "Editor" part of the name indicates that it's there to edit fields which this widget actually does not) , I would inherit directly from QObject